### ISSUE TITLE
Using format: uri leads to not compilable code 

### ISSUE THREAD

**Issue by ok11** [go-swagger/go-swagger#1769]
## Problem statement

If one uses a construct like 
```
type: string
format: uri
```
is a schema definition, the generated operation is missing `strfmt `import, hence not compilable

## Swagger specification
```
paths:
  /users:
    post:
      tags:
      - User
      summary: "an operation to register a new user"
      description: >-
        Registers a new user
      operationId: "registerUser"
      consumes:
      - "application/json"
      produces:
      - "application/json"
      parameters:
      - name: payload
        in: body
        description: >-
          Data for the user to be registered
        schema:
          $ref: "./Data.yaml"
        required: true
      responses:
        201:
          description: Created
          headers:
            Location:
              description: URI of the registered resource
              type: string
              **format: uri**
            ETag:
              description: >-
                Fingerprint of the created resource
              type: string
          schema:
            $ref: "./User.yaml"
```

Generated operation:
```
package user

// This file was generated by the swagger tool.
// Editing this file might prove futile when you re-run the swagger generate command

import (
	"net/http"

	"github.com/go-openapi/runtime"

	dto "github.wdf.sap.corp/bb-go/userservice/pkg/api/rest/gen/dto"
)

// RegisterUserCreatedCode is the HTTP code returned for type RegisterUserCreated
const RegisterUserCreatedCode int = 201

/*RegisterUserCreated Created

swagger:response registerUserCreated
*/
type RegisterUserCreated struct {
	/*Fingerprint of the created resource

	 */
	ETag string `json:"ETag"`
	/*URI of the registered resource

	 */
	Location **strfmt.URI** `json:"Location"`

	/*
	  In: Body
	*/
	Payload *dto.User `json:"body,omitempty"`
}

```
As you see the generated code uses `strfmt.URI`, while `strfmt` is not imported

## Steps to reproduce

Just generate...

## Environment
swagger version: 0.17.0 
go version: 1.11  
OS:  Windows 10 (Linux subsystem/Ubuntu 18.04)

**Comment by ok11** [comment]
Hi, thanks for the quick reply!

It works fine in all other cases, e.g. see below:

```
import (
	"net/http"

	"github.com/go-openapi/errors"
	"github.com/go-openapi/runtime/middleware"

	strfmt "github.com/go-openapi/strfmt"
)

// NewDeleteUserParams creates a new DeleteUserParams object
// no default values defined in spec.
func NewDeleteUserParams() DeleteUserParams {

	return DeleteUserParams{}
}
```
The file has been generated in the same generation approach. `strsfmt` is added in `go.mod`  and works fine in all other cases. So far only URI makes problems, as far as I can conclude.

**Comment by ok11** [comment]
... and commenting out `format: uri` works around the problem (since `strfmt` is not involved anymore)

**Comment by fredbi** [comment]
Sorry but I tried to reproduce (btw, your spec cannot be reused as is) and this generated correct import with URI.

**Comment by ok11** [comment]
Ok, sorry about exposing only a fragment. Let me attach the entire spec and the commad line options. The only thing is that I heavily modularize the API definition and hence use `$ref` intensively, so it is not a one file, but a tree. I created a gist, which flattened a bit the structure, but still contains all the important parts: [gist](https://gist.github.com/ok11/6366b620192278b677cba169fc4c6db2)

The command line is:

`swagger generate server --spec=pkg/api/rest/def/api.yaml --model-package=dto --template=stratoscale --template-dir=pkg/api/rest/templates --name=UserService --target=pkg/api/rest/gen --exclude-main`

**Comment by ok11** [comment]
Yes, and `format:uri` is currently commented out in the gist as I just wanted to get it compiled

**Comment by fredbi** [comment]
suddenly a lot of relevant details appear:
- you are using stratoscale templates
- you are overriding this with your own templates

Generation works well uri or not uri with standard options. So this most come from one of the above settings

**Comment by ok11** [comment]
BTW, is reporting a bug on stratoscale in your repo a correct approach or I should rather go via theirs? Not sure what is your collaboration process for templates contribution..

**Comment by fredbi** [comment]
afaik @posener wanted this repo to be the main source of maintenance for their template.

If you find bugs in this, feel free to send a PR here

**Comment by fredbi** [comment]
@ok11 are you still facing this problem with latest master?

**Comment by ok11** [comment]
Hey @fredbi,  sorry, didn't give it a try -- have been completely buried at work since a couple of weeks. But thanks for reminding, will do it tonight.

**Comment by fredbi** [comment]
Thanks. You see, I am the broken window fixer here... So I am making sure no window remains broken for too long.

**Comment by ok11** [comment]
Tried 0.17.2 -- no luck. But I also did what I should have done when you said you use `goimports`. So I just ran` goimports `on the generated code and it failed to add `strfmt` as well. In other words, not an issue of `go-swagger`. May be a sort of maturity issue of `goimports` in conjunction with `go mod`. In any case, if you prefer to close the issue, please do so. If you keep it open for tracking purposes, also fine with me, will try to investigate further and document the results here.

**Comment by fredbi** [comment]
ok thanks for the feedback. It is another failure with go mods, and yes probably goimports.
That must be what they call "rough edges" in go11 release notes... 

Let's keep this one open, so we'll figure out when tackling issues with modules (not my current priority)

**Comment by fredbi** [comment]
Anyway, just FYI, there is a way to really make sure: instead of default config for go-swagger, override the swagger.yaml and tell the generator to skip the formatting. Then you'll have raw generation.
Normally you should see all your imports here (possibly even duplicates). So if this works well, the the formatting is the culprit. This includes goimports.
